hoomd/compute.py

```diff
@@ -12,139 +12,34 @@
 
-    def _connect_gsd(self, gsd):
-        # This is an internal method, and should not be called directly. See gsd.dump_state() instead
-        if isinstance(gsd, hoomd.dump.gsd) and hasattr(self.cpp_compute, "connectGSDStateSignal"):
-            self.cpp_compute.connectGSDStateSignal(gsd.cpp_analyzer, self._gsd_state_name());
-        else:
-            raise NotImplementedError("GSD Schema is not implemented for {}".format(self.__class__.__name__));
-
-    def restore_state(self):
-        """ Restore the state information from the file used to initialize the simulations
-        """
-        if isinstance(hoomd.context.current.state_reader, _hoomd.GSDReader) and hasattr(self.cpp_compute, "restoreStateGSD"):
-            self.cpp_compute.restoreStateGSD(hoomd.context.current.state_reader, self._gsd_state_name());
-        else:
-            if hoomd.context.current.state_reader is None:
-                hoomd.context.current.device.cpp_msg.error("Can only restore after the state reader has been initialized.\n");
-            else:
-                hoomd.context.current.device.cpp_msg.error("Restoring state from {reader_name} is not currently supported for {name}\n".format(reader_name=hoomd.context.current.state_reader.__name__, name=self.__class__.__name__));
-            raise RuntimeError("Can not restore state information!");
 
-
-# set default counter
-_compute.cur_id = 0;
-
-class thermo(_compute):
+class ThermoQuantities(_Thermo):
     R""" Compute thermodynamic properties of a group of particles.
 
     Args:
-        group (``hoomd.group``): Group to compute thermodynamic properties for.
+        filter (``hoomd.filter``): Particle filter to compute thermodynamic properties for.
 
-    :py:class:`hoomd.compute.thermo` acts on a given group of particles and calculates thermodynamic properties of those particles when
-    requested. A default :py:class:`hoomd.compute.thermo` is created that operates on the group of all particles. Integration methods
-    such as ``hoomd.md.integrate.nvt`` automatically create an internal :py:class:`hoomd.compute.thermo` for the group that they operate on.
-    If thermodynamic properties are needed on additional groups, a user can specify additional :py:class:`hoomd.compute.thermo` commands.
+    :py:class:`hoomd.compute.ThermoQuantities` acts on a given group of particles and calculates thermodynamic properties of those particles when
+    requested. A default :py:class:`hoomd.compute.ThermoQuantities` is created that operates on the group of all particles. Integration methods
+    such as ``hoomd.md.integrate.NVT`` automatically create an internal :py:class:`hoomd.compute.ThermoQuantities` for the group that they operate on.
+    If thermodynamic properties are needed on additional groups, a user can specify additional :py:class:`hoomd.compute.ThermoQuantities` commands.
 
     Whether they are automatically created or created by a user, all specified thermos are available for logging via
```

Comment:
Really, this whole documentation just needs to be re-written. It probably just needs a few sentences that state the class computes thermodynamic quantities and interoperates with the `Logger`.

Reply:
I think we should keep all the documentation related to the quantities and the put them underneath their properties. The class documentation can be shortened significantly.